packages/sil-voting-app/src/routes/dashboard/components/Proposal/Description.tsx

```diff
@@ -0,0 +1,51 @@
+import Collapse from '@material-ui/core/Collapse';
+import Block from 'medulas-react-components/lib/components/Block';
+import Typography from 'medulas-react-components/lib/components/Typography';
+import React, { useState } from 'react';
+import { elipsify } from '../../../../utils/strings';
+
+const DESC_MAX_LENGTH = 180;
+
+interface Props {
+  readonly description: string;
+}
+
+const Description = (props: Props): JSX.Element => {
+  const [expanded, setExpanded] = useState(false);
+
+  const onClick = (): void => {
+    setExpanded(!expanded);
+  };
+
+  return (
+    <Block marginBottom={1}>
+      {props.description.length < DESC_MAX_LENGTH && (
```

Comment:
Seeing this component does not have a big hierarchy pending from it I would do it more readable:

```
const smallDescription = props.description.length < DESC_MAX_LENGTH
if (smallDescription) {
  return <Block marginBottom={1}><Typography variant="body1">{props.description}</Typography></Block>
}

if (!expanded) {
   return (
       <Block marginBottom={1}>
          <Typography inline variant="body1">
              {elipsify(props.description, DESC_MAX_LENGTH)}{' '}
          </Typography>
          <Typography inline link onClick={onClick} variant="body1" weight="semibold">
              Read more
          </Typography>
      </Block>
    )
}

return (
       <Block marginBottom={1}>
          <Collapse in={expanded} timeout="auto" unmountOnExit>
            <Typography inline variant="body1">
              {props.description}{' '}
            </Typography>
            <Typography inline link onClick={onClick} variant="body1" weight="semibold">
              Read less
            </Typography>
          </Collapse>
      </Block>
    )
```

In this way, we can understand what the component returns based on its internal state in less than one second.

Reply:
I agree with the simplification. However, this code makes the `Collapse` animation disappear. The rendering is immediate when clicking "Read more" or "Read less".

Reply:
Ahhh, I see. Good point. In this case I suggest:
```
return (
    <Block marginBottom={1}>
        { !expanded && <ReadMore .../> }
         <Collapse in={expanded} timeout="auto" unmountOnExit>
            <Typography inline variant="body1">
              {props.description}{' '}
            </Typography>
            <Typography inline link onClick={onClick} variant="body1" weight="semibold">
              Read less
            </Typography>
         </Collapse>
    </Block>
```

And `<ReadMore />` component should be inside the same class, meaning Description.tsx, there is no need to create a separate component, you can inline it as a variable.